Workaround fix for TOC of Deploy for production page (#1244)
redpanda-data/documentation: modules/reference/pages/properties/cluster-properties.adoc

---

=== consumer_offsets_topic_batch_cache_enabled

> This property lets you enable the batch cache for the consumer offsets topic. By default, the cache for consumer offsets topic is disabled. Changing this property is not recommended in production systems, as it may affect performance. The change is applied only after the restart.

Reviewer: ```suggestion
This property lets you enable batch caching for the consumer offsets topic. By default, the cache for this topic is disabled. Changing this property is not recommended in production systems as it may affect performance.
```

---

=== datalake_disk_space_monitor_enable

> Option to explicitly disable enforcement of datalake disk space usage.

Reviewer: The config name is confusing 😓 Does this mean you can disable disk space usage limits for the datalake?

Also what is the datalake?

Reviewer: So you set this to `false` to disable? In that case, which configs are ignored?

*Requires restart:* No

*Visibility:* `user`

*Type:* boolean

*Default:* `true`

---

=== datalake_disk_usage_overage_coeff

The datalake disk usage monitor reclaims the overage multiplied by this this coefficient to compensate for data that is written during the idle period between control loop invocations.

*Requires restart:* No

*Visibility:* `tunable`

*Type:* number

*Default:* `2.0`

---

=== datalake_scheduler_disk_reservation_block_size

The size, in bytes, of the block of disk reservation that the datalake manager will assign to each datalake scheduler when it runs out of local reservation.

*Requires restart:* No

*Visibility:* `tunable`

*Type:* integer

*Default:* `50_mib`

---

=== datalake_scratch_space_size_bytes

Size, in bytes, of the amount of scratch space datalake should use.

*Unit:* bytes

*Requires restart:* No

*Visibility:* `tunable`

*Type:* integer

*Default:* `5_gib`

---

=== datalake_scratch_space_soft_limit_size_percent

> Size of the scratch space datalake soft limit expressed as a percentage of the datalake_scratch_space_size_bytes configuration value.

Reviewer 2: ```suggestion
Size of the scratch space datalake soft limit expressed as a percentage of the `datalake_scratch_space_size_bytes` configuration value.
```

// end::http_authentication[]

---

=== iceberg_backlog_controller_i_coeff

> Integral coefficient for the Iceberg backlog controller. The error is integrated (accumulated) over time and the aggregated value contributes to the datalake translation priority with this coefficient.

Reviewer: Could we simplify this a bit?

This setting controls how much past backlog (unprocessed work) affects the priority of processing new data in the Iceberg system. It uses an integral coefficient, which means it adds up backlog errors over time, and this total influences how urgently the system works on translating data, based on the value you set here.

Reviewer 2: ```suggestion
Controls how much past backlog (unprocessed work) affects the priority of processing new data in the Iceberg system. The system accumulates backlog errors over time, and this coefficient determines how much that accumulated backlog influences the urgency of data translation.
```

// end::iceberg_invalid_record_action[]

=== iceberg_latest_schema_cache_ttl_ms

> The TTL for the cache in translation that stores the latest schema when using the `latest_protobuf_value` iceberg mode.

Reviewer 2: There's no mention of this value in other doc pieces

Reviewer: maybe add something like 'This setting allows users to control how long the latest schema is cached during translation, impacting schema refresh behavior and performance.'

Reviewer 2: ```suggestion
The TTL for caching the latest schema during translation when using the `latest_protobuf_value` iceberg mode. This setting controls how long the latest schema remains cached during translation, which affects schema refresh behavior and performance.
```

---

=== iceberg_rest_catalog_aws_access_key

> AWS access key for Iceberg REST catalog SigV4 authentication. If not set, falls back to cloud_storage_access_key when using aws_sigv4 authentication mode.

Reviewer: ```suggestion
AWS access key for Iceberg REST catalog SigV4 authentication. If not set, falls back to the value of `cloud_storage_access_key` when using aws_sigv4 authentication mode.
```

We should link to that `cloud_storage_access_key` config

*Requires restart:* Yes

*Visibility:* `user`

*Type:* string

*Default:* `null`

---

=== iceberg_rest_catalog_credentials_source

> Source of AWS credentials for Iceberg REST catalog SigV4 authentication. If not set, falls back to cloud_storage_credentials_source when using aws_sigv4 authentication mode. Accepted values: config_file, aws_instance_metadata, sts, gcp_instance_metadata, azure_vm_instance_metadata, azure_aks_oidc_federation.

Reviewer: same as the comment above. Also use inline code format for configs.

Reviewer: apply to all config descriptions.

*Requires restart:* Yes

*Visibility:* `user`

*Default:* `null`

---

=== iceberg_rest_catalog_aws_region

AWS region for Iceberg REST catalog SigV4 authentication. If not set, falls back to cloud_storage_region when using aws_sigv4 authentication mode.

*Requires restart:* Yes

*Visibility:* `user`

*Type:* string

*Default:* `null`

---

=== iceberg_rest_catalog_aws_secret_key

AWS secret key for Iceberg REST catalog SigV4 authentication. If not set, falls back to cloud_storage_secret_key when using aws_sigv4 authentication mode.

*Requires restart:* Yes

*Visibility:* `user`

*Type:* string

*Default:* `null`

---

=== iceberg_rest_catalog_aws_service_name

AWS service name for SigV4 signing when using aws_sigv4 authentication mode. Defaults to 'glue' for AWS Glue Data Catalog. Can be changed to support other AWS services that provide Iceberg REST catalog APIs.

*Requires restart:* Yes

*Visibility:* `user`

*Type:* string

*Default:* `glue`

---

=== iceberg_rest_catalog_base_location

> Base URI for the Iceberg REST catalog. If unset, the REST catalog server determines the location. Some REST catalogs, like AWS Glue, require the client to set this. After Iceberg is enabled, do not change this value.

Reviewer: > After Iceberg is enabled, do not change this value.

or what? 😅

---

=== iceberg_rest_catalog_warehouse

> Warehouse to use for the Iceberg REST catalog. Redpanda will query the catalog for configurations specific to the warehouse, for example, using it to automatically configure the appropriate prefix.

Reviewer: ```suggestion
Warehouse to use for the Iceberg REST catalog. Redpanda queries the catalog for configurations specific to the warehouse, for example, using it to automatically configure the appropriate prefix.
```

---

> === raft_replica_max_flush_delay_ms

Reviewer: deprecated?

Reviewer 2: No. It was duplicated in this page.

// end::sasl_mechanisms[]

---

=== schema_registry_always_normalize

> Always normalize schemas. If set, this overrides the normalize parameter in API requests.

Reviewer: ```suggestion
Always normalize schemas. If set, this overrides the `normalize` parameter in requests to the Schema Registry API.
```
